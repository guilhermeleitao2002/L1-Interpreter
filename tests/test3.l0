let add = fn x, y => {
  x + y
};

let mul = fn x, y => {
  x * y
};

let result1 = ((add) (5, 3));
let result2 = ((mul) (4, 6));
println result1;
println result2;

let compose = fn f, g, x => {
  ((f) (((g) (x, 2)), 3))
};

let result3 = ((compose) (add, mul, 5));
println result3;;