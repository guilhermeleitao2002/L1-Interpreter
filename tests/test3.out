8
24
13